### PR TITLE
display placeholder for rich-text-widget

### PR DESCRIPTION
**Pull request by ETLaurent**
Depends on #3882
Closes [PRO-3146](https://linear.app/apostrophecms/issue/PRO-3146/implement-per-area-and-global-fallback-placeholder-options-for-rich)

## Summary

Set a default placeholder value for rich text widget:

image

The placeholder disappears as soon as we start typing in.
It's overridable on project side, so if it's overridden with a falsy value, it will display the normal/previous behaviour.

## What are the specific steps to test this change?

On testbed:

```js
// in a project's modules/@apostrophecms/rich-text-widget/index.js
options: {
    placeholderText: 'Start typing here!!!'
  },
```

👇 

image

---

```js
// in a project's modules/@apostrophecms/rich-text-widget/index.js
options: {
    placeholderText: false // or any falsy value
  },
```

👇 

image

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/AposRichTextWidgetEditor.vue [hunk 1]
```diff
@@ -336,6 +354,15 @@ export default {
...
   }
 
+  .apos-rich-text-editor__editor ::v-deep .ProseMirror p.is-empty:first-child::before {
+    content: attr(data-placeholder);
+    float: left;
+    color: #adb5bd;
+    pointer-events: none;
+    height: 0;
+    margin-left: 5px;
+  }
```
Comment (ETLaurent) [inline modules/@apostrophecms/rich-text-widget/ui/apos/components/AposRichTextWidgetEditor.vue:364]: see https://embed.tiptap.dev/preview/Extensions/Placeholder

File: modules/@apostrophecms/rich-text-widget/index.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ module.exports = {
     icon: 'format-text-icon',
     label: 'apostrophe:richText',
     contextual: true,
+    placeholderText: 'apostrophe:richTextPlaceholder',
```
Comment (ETLaurent) [inline modules/@apostrophecms/rich-text-widget/index.js:12]: @stuartromanek if we want to set a default value, looks like this:

image

It is overridable on project side, so if a project sets `placeholderText: false` or any falsy value, it'll go back to the previous behavior:

image

Comment (stuartromanek) [reply modules/@apostrophecms/rich-text-widget/index.js:12]: I think we should keep a default in core.

'Start typing here...' is fine

**Review by stuartromanek — CHANGES_REQUESTED** [review]
Two things standing out to me:
- A rich text widget should be able to override the module defaults and provide its own placeholder text via the RT options object. `placeholder` is probably a fine key
- The placeholder text should only be displayed when the widget is active and focused. This would happen when first adding the RT widget to the area or clicking in to focus an existing, empty RT widget. When the RT widget is empty and unfocused it should return to the previous A3 empty state.

**Review by stuartromanek — APPROVED** [review]

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/AposRichTextWidgetEditor.vue [hunk 3]
```diff
@@ -176,19 +186,63 @@ export default {
...
-        Underline
-      ].concat(this.aposTiptapExtensions)
+      extensions,
+
+      // The following events are triggered:
+      //  - before the placeholder configuration function, when loading the page
+      //  - after it, once the page is loaded and we interact with the editors
+      // To solve this issue, use another `this.showPlaceholder` variable
+      // and toggle it after the placeholder configuration function is called,
+      // thanks to textTick.
```
Comment (boutell) [inline modules/@apostrophecms/rich-text-widget/ui/apos/components/AposRichTextWidgetEditor.vue:230]: nextTick

**Review by boutell — CHANGES_REQUESTED** [review]
Very minor notes

**Review by boutell — APPROVED** [review]

**Pull request merged**